Cython/Compiler/Parsing.py

```diff
@@ -3133,31 +3133,63 @@ def p_cdef_extern_block(s, pos, ctx):
 def p_c_enum_definition(s, pos, ctx):
     # s.sy == ident 'enum'
     s.next()
+
+    scoped = False
+
+    if s.sy == 'class' or (s.sy == 'IDENT' and s.systring == 'struct'):
```

Comment:
I think we can at least reduce the impact by limiting this to C++ mode:
```suggestion
    if s.context.cpp and (s.sy == 'class' or (s.sy == 'IDENT' and s.systring == 'struct')):
```
For C++ users, it will be understandable that `cdef enum struct` isn't a valid struct name, but C users will not have to care.